### PR TITLE
Refactor(plugins): Move jinja filter code for arista.avd.is_in_filter to PyAVD

### LINKED ISSUE #4132: Refactor(plugins): Move jinja filter code for arista.avd.is_in_filter to PyAVDFeature request Title
### Enhancement summary

Move jinja filter code for arista.avd.is_in_filter to PyAVD

### Which component of AVD is impacted

plugins

### Use case example

.

### Describe the solution you would like

The code and unit tests for is_in_filter filter should be moved to pyavd/j2filters and imported inside plugins/filter using pyavd wrapper.

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Move jinja filter code for arista.avd.is_in_filter to PyAVD

## Related Issue(s)

Fixes #4132 

## Component(s) name

`arista.avd.is_in_filter`

## Proposed changes
Moving the code and unit tests for arista.avd.is_in_filter filter to PyAVD

## How to test
CI should catch things.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM

File: python-avd/tests/pyavd/j2filters/test_is_in_filter.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+HOSTNAME_INVALID = "test1.bsn.com"
+HOSTNAME_FILTER_VALID = ["arista", "aristanetworks"]
+HOSTNAME_FILTER_INVALID = "aristanetworks"
+
+IS_IN_FILTER_TEST_CASES = [
+    # (<hostname>, <hostname_filter>, <res_is_in_filter>)
+    (HOSTNAME_VALID, None, True),  # default filter
+    (HOSTNAME_VALID, HOSTNAME_FILTER_VALID, True),  # valid hostname
+    (HOSTNAME_INVALID, HOSTNAME_FILTER_VALID, False),  # invalid hostname
+    # TODO: Check if this is a valid testcase. Add a type check?
```
Comment (gmuloc) [inline python-avd/tests/pyavd/j2filters/test_is_in_filter.py:19]: This should not be working like this for the last example. Can you please log an issue to revisit the behavior of this filter after migration as changing this now would be a breaking change probably (here we end up looping over every character in the string and checking if any character is in the hostname, which is not the spirit of the filter, or at least should not be given like this as an input)

**Review by gmuloc — APPROVED** [review]
Approving the move. Some of the filter behavior is dubious but that's not the purpose of this PR

File: python-avd/tests/pyavd/j2filters/test_is_in_filter.py (hunk 1 above)
Comment (ClausHolbechArista) [reply python-avd/tests/pyavd/j2filters/test_is_in_filter.py:19]: There is some historic behavior here, where the device-filters in the cvp-related roles accepted a string or a list.

**Pull request merged**